io/src/main/java/com/topjohnwu/superuser/internal/FifoOutputStream.java

```diff
@@ -74,16 +73,17 @@ class FifoOutputStream extends FilterOutputStream {
         }
     }
 
-    private boolean checkFile(SuFile file) {
+    private void checkFile(SuFile file) throws FileNotFoundException {
         if (file.isDirectory())
-            return false;
+            throw new FileNotFoundException(file.getAbsolutePath() + " is not a file but a directory");
         if (file.isBlock() || file.isCharacter()) {
             append = false;
-            return true;
         }
-        if (append)
-            return file.canWrite() || file.createNewFile();
-        return file.clear();
+        if (append && !file.canWrite() && !file.createNewFile()) {
+            throw new FileNotFoundException("Can write to file "+file.getAbsolutePath());
```

Comment:
Can you add spaces between the strings and `+`?

Reply:
Sure, it's done.
(I avoid calling IDE formatting feature on code base that is not mine)